lib/core/src/model.rs

```diff
@@ -586,6 +588,11 @@ pub struct Payment {
     /// In case of a Send swap, this is the preimage of the paid invoice (proof of payment).
     pub preimage: Option<String>,
 
+    /// Represents the invoice associated with a payment
+    /// In the case of a Send payment, this is the invoice paid by the swapper
+    /// In the case of a Receive payment, this is the invoice paid by the user
+    pub invoice: Option<String>,
```

Comment:
Why is it optional? Is it because of future support of onchain swaps and different kind of payments?